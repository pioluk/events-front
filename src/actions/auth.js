/* @flow */

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from '../constants/auth'

export function loginRequest (username: string, password: string) {
  return {
    type: LOGIN_REQUEST,
    username,
    password
  }
}

export function loginSuccess (user: string, token: string) {
  return {
    type: LOGIN_SUCCESS,
    user,
    token
  }
}

export function loginFailure (error: Error) {
  return {
    type: LOGIN_FAILURE,
    error
  }
}

export function logout () {
  return {
    type: LOGOUT
  }
}
