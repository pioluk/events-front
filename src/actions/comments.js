// @flow

import {
  FETCH_COMMENTS,
  FETCH_COMMENTS_SUCCESS,
  FETCH_COMMENTS_FAILURE,
  ADD_COMMENT,
  ADD_COMMENT_SUCCESS,
  ADD_COMMENT_FAILURE,
  REMOVE_COMMENT,
  REMOVE_COMMENT_SUCCESS,
  REMOVE_COMMENT_FAILURE,
} from '../constants/comments'

export function fetchComments (eventId: number, limit: number, offset: number) {
  return {
    type: FETCH_COMMENTS,
    eventId,
    limit,
    offset
  }
}

export function fetchCommentsSuccess (count: number, comments: any[]) {
  return {
    type: FETCH_COMMENTS_SUCCESS,
    count,
    comments
  }
}

export function fetchCommentsFailure (error: Error) {
  return {
    type: FETCH_COMMENTS_FAILURE,
    error
  }
}

export function addComment (text: string) {
  return {
    type: ADD_COMMENT,
    text
  }
}

export function addCommentSuccess (comment: any) {
  return {
    type: ADD_COMMENT_SUCCESS,
    comment
  }
}

export function addCommentFailure (error: Error) {
  return {
    type: ADD_COMMENT_FAILURE,
    error
  }
}

export function removeComment (id: number) {
  return {
    type: REMOVE_COMMENT,
    id
  }
}

export function removeCommentSuccess (id: number) {
  return {
    type: REMOVE_COMMENT_SUCCESS,
    id
  }
}

export function removeCommentFailure (error: Error) {
  return {
    type: REMOVE_COMMENT_FAILURE,
    error
  }
}
