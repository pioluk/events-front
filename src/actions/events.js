import {
  FETCH_EVENTS_REQUEST,
  FETCH_EVENTS_SUCCESS,
  FETCH_EVENTS_FAILURE
} from '../constants/events'

export function fetchEventsRequest () {
  return {
    type: FETCH_EVENTS_REQUEST
  }
}

export function fetchEventsSuccess (events) {
  return {
    type: FETCH_EVENTS_SUCCESS,
    events
  }
}

export function fetchEventsFailure (error) {
  return {
    type: FETCH_EVENTS_FAILURE,
    error
  }
}
