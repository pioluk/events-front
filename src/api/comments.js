// @flow

import { apiGet, apiPost, apiDelete } from './utils'
import { COMMENT_PAGE_SIZE } from '../config'

export const getComments = (eventId: number, page: number) =>
  apiGet(`event/${eventId}/comment?limit=${COMMENT_PAGE_SIZE}&offset=${(page - 1) * COMMENT_PAGE_SIZE}`)

export const createComment = (eventId: number, text: string) =>
  apiPost(`event/${eventId}/comment`, { text })

export const removeComment = (eventId: number, commentId: number) =>
  apiDelete(`event/${eventId}/comment/${commentId}`)
