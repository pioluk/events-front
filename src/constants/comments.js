export const FETCH_COMMENTS = 'FETCH_COMMENTS'
export const FETCH_COMMENTS_SUCCESS = 'FETCH_COMMENTS_SUCCESS'
export const FETCH_COMMENTS_FAILURE = 'FETCH_COMMENTS_FAILURE'
export const ADD_COMMENT = 'ADD_COMMENT'
export const ADD_COMMENT_SUCCESS = 'ADD_COMMENT_SUCCESS'
export const ADD_COMMENT_FAILURE = 'ADD_COMMENT_FAILURE'
export const REMOVE_COMMENT = 'REMOVE_COMMENT'
export const REMOVE_COMMENT_SUCCESS = 'REMOVE_COMMENT_SUCCESS'
export const REMOVE_COMMENT_FAILURE = 'REMOVE_COMMENT_FAILURE'
export const INCREASE_PAGE = 'INCREASE_PAGE'
export const RESET_COMMENTS = 'RESET_COMMENTS'
