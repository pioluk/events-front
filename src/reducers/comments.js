// @flow

import {
  FETCH_COMMENTS,
  FETCH_COMMENTS_SUCCESS,
  FETCH_COMMENTS_FAILURE,
  ADD_COMMENT,
  ADD_COMMENT_SUCCESS,
  ADD_COMMENT_FAILURE,
  REMOVE_COMMENT,
  REMOVE_COMMENT_SUCCESS,
  REMOVE_COMMENT_FAILURE,
  RESET_COMMENTS
} from '../constants/comments'

type Comment = any

type CommentsState = {
  totalCount: number,
  commentCount: number,
  comments: Array<Comment>,
  error: ?Error,
  isLoading: boolean,
}

const initialState: CommentsState = {
  totalCount: 0,
  commentCount: 0,
  comments: [],
  error: null,
  isLoading: false,
}

export default function commentsReducer (state: CommentsState = initialState, action: any) {
  switch (action.type) {
    case FETCH_COMMENTS:
      return { ...state, isLoading: true }

    case FETCH_COMMENTS_SUCCESS:
      return {
        ...state,
        totalCount: action.count,
        commentCount: state.commentCount + action.comments.length,
        comments: [...state.comments, ...action.comments],
        isLoading: false
      }

    case FETCH_COMMENTS_FAILURE:
      return { ...state, error: action.error, isLoading: false }

    case ADD_COMMENT:
      return { ...state }

    case ADD_COMMENT_SUCCESS:
      return {
        ...state,
        comments: [action.comment, ...state.comments]
      }

    case ADD_COMMENT_FAILURE:
      return { ...state, error: action.error }

    case REMOVE_COMMENT_SUCCESS: {
      const comments = state.comments.filter(comment => comment.id !== action.id)
      return { ...state, comments }
    }

    case REMOVE_COMMENT_FAILURE:
      return { ...state, error: action.error }

    case RESET_COMMENTS:
      return initialState

    default:
      return state
  }
}
