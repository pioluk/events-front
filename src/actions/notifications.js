import { DISPLAY_NOTIFICATION } from '../constants/notifications'

export function displayNotification (message) {
  return {
    type: DISPLAY_NOTIFICATION,
    message
  }
}
