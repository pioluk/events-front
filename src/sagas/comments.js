import { call, take, put } from 'redux-saga/effects'
import { FETCH_COMMENTS, ADD_COMMENT, REMOVE_COMMENT } from '../constants/comments'
import * as actions from '../actions/comments'
import { getComments, createComment, removeComment } from '../api/comments'

export function* getCommentsFlow () {
  while (true) {
    const { eventId, limit, offset } = yield take(FETCH_COMMENTS)
    try {
      const { count, comments } = yield call(getComments, eventId, limit, offset)
      if (comments) {
        yield put(actions.fetchCommentsSuccess(count, comments))
      }
    } catch (err) {
      yield put(actions.fetchCommentsFailure(err))
    }
  }
}

export function* createCommentFlow () {
  while (true) {
    const { eventId, text } = yield take(ADD_COMMENT)
    try {
      const comment = yield call(createComment, eventId, text)
      if (comment) {
        yield put(actions.addCommentSuccess(comment))
      }
    } catch (err) {
      yield put(actions.addCommentFailure(err))
    }
  }
}

export function* removeCommentFlow () {
  while (true) {
    const { id } = yield take(REMOVE_COMMENT)
    try {
      yield call(removeComment, id)
      yield put(actions.removeCommentSuccess(id))
    } catch (err) {
      yield put(actions.removeCommentFailure(err))
    }
  }
}
