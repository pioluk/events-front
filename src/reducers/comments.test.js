// global describe,it,expect

import reducer from './comments'
import * as actions from '../actions/comments'

describe('comments reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle FETCH_COMMENTS', () => {
    const state = reducer(undefined, actions.fetchComments(1, 20, 0))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_COMMENTS_SUCCESS', () => {
    const comments = [
      {
        user: {
          id: 1,
          username: 'pioluk',
          email: '181423@edu.p.lodz.pl',
          imageAvatar: null,
          createdAt: '2016-12-01T13:02:38.213Z',
          updatedAt: '2016-12-01T13:02:38.213Z'
        },
        id: 1,
        UserId: 1,
        EventId: 1,
        text: 'Lorem ipsum',
        createdAt: '2016-12-03T12:00:00.000Z',
        updatedAt: '2016-12-03T12:00:00.000Z',
        deletedAt: null
      },
      {
        user: {
          id: 1,
          username: 'pioluk',
          email: '181423@edu.p.lodz.pl',
          imageAvatar: null,
          createdAt: '2016-12-01T13:02:38.213Z',
          updatedAt: '2016-12-01T13:02:38.213Z'
        },
        id: 2,
        UserId: 1,
        EventId: 1,
        text: 'Dolor Amet',
        createdAt: '2016-12-03T12:30:00.000Z',
        updatedAt: '2016-12-03T12:30:00.000Z',
        deletedAt: null
      }
    ]
    const state = reducer(undefined, actions.fetchCommentsSuccess(comments))
    expect(state).toMatchSnapshot()
  })

  it('should handle FETCH_COMMENTS_FAILURE', () => {
    const error = new Error('Error fetching comments')
    const state = reducer(undefined, actions.fetchCommentsFailure(error))
    expect(state).toMatchSnapshot()
  })

  it('should handle ADD_COMMENT', () => {
    const text = 'Comment #1'
    const state = reducer(undefined, actions.addComment(text))
    expect(state).toMatchSnapshot()
  })

  it('should handle ADD_COMMENT_SUCCESS', () => {
    const comment = {
      user: {
        id: 1,
        username: 'pioluk',
        email: '181423@edu.p.lodz.pl',
        imageAvatar: null,
        createdAt: '2016-12-01T13:02:38.213Z',
        updatedAt: '2016-12-01T13:02:38.213Z'
      },
      id: 1,
      UserId: 1,
      EventId: 1,
      text: 'Lorem ipsum',
      createdAt: '2016-12-03T12:00:00.000Z',
      updatedAt: '2016-12-03T12:00:00.000Z',
      deletedAt: null
    }
    const state = reducer(undefined, actions.addCommentSuccess(comment))
    expect(state).toMatchSnapshot()
  })

  it('should handle ADD_COMMENT_FAILURE', () => {
    const error = new Error('Error adding comment')
    const state = reducer(undefined, actions.addCommentFailure(error))
    expect(state).toMatchSnapshot()
  })

  it('should handle REMOVE_COMMENT', () => {
    const id = 1
    const state = reducer(undefined, actions.removeComment(id))
    expect(state).toMatchSnapshot()
  })

  it('should handle REMOVE_COMMENT_SUCCESS', () => {
    const id = 1
    const initialState = {
      comments: [
        {
          user: {
            id: 1,
            username: 'pioluk',
            email: '181423@edu.p.lodz.pl',
            imageAvatar: null,
            createdAt: '2016-12-01T13:02:38.213Z',
            updatedAt: '2016-12-01T13:02:38.213Z'
          },
          id: 1,
          UserId: 1,
          EventId: 1,
          text: 'Lorem ipsum',
          createdAt: '2016-12-03T12:00:00.000Z',
          updatedAt: '2016-12-03T12:00:00.000Z',
          deletedAt: null
        },
        {
          user: {
            id: 1,
            username: 'pioluk',
            email: '181423@edu.p.lodz.pl',
            imageAvatar: null,
            createdAt: '2016-12-01T13:02:38.213Z',
            updatedAt: '2016-12-01T13:02:38.213Z'
          },
          id: 2,
          UserId: 1,
          EventId: 1,
          text: 'Dolor Amet',
          createdAt: '2016-12-03T12:30:00.000Z',
          updatedAt: '2016-12-03T12:30:00.000Z',
          deletedAt: null
        }
      ],
      error: null,
      isLoading: false
    }
    const state = reducer(initialState, actions.removeCommentSuccess(id))
    expect(state).toMatchSnapshot()
  })

  it('should handle REMOVE_COMMENT_FAILURE', () => {
    const error = new Error('Error removing comment')
    const state = reducer(undefined, actions.removeCommentFailure(error))
    expect(state).toMatchSnapshot()
  })
})
