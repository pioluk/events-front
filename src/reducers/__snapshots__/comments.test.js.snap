exports[`comments reducer should handle ADD_COMMENT 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": null,
  "isLoading": false,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle ADD_COMMENT_FAILURE 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": [Error: Error adding comment],
  "isLoading": false,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle ADD_COMMENT_SUCCESS 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [
    Object {
      "EventId": 1,
      "UserId": 1,
      "createdAt": "2016-12-03T12:00:00.000Z",
      "deletedAt": null,
      "id": 1,
      "text": "Lorem ipsum",
      "updatedAt": "2016-12-03T12:00:00.000Z",
      "user": Object {
        "createdAt": "2016-12-01T13:02:38.213Z",
        "email": "181423@edu.p.lodz.pl",
        "id": 1,
        "imageAvatar": null,
        "updatedAt": "2016-12-01T13:02:38.213Z",
        "username": "pioluk",
      },
    },
  ],
  "error": null,
  "isLoading": false,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle FETCH_COMMENTS 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": null,
  "isLoading": true,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle FETCH_COMMENTS_FAILURE 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": [Error: Error fetching comments],
  "isLoading": false,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle FETCH_COMMENTS_SUCCESS 1`] = `
Object {
  "commentCount": 2,
  "comments": Array [
    Object {
      "EventId": 1,
      "User": Object {
        "id": 1,
        "imageAvatar": null,
        "username": "pioluk",
      },
      "UserId": 1,
      "createdAt": "2016-12-03T12:00:00.000Z",
      "deletedAt": null,
      "id": 1,
      "text": "Lorem ipsum",
      "updatedAt": "2016-12-03T12:00:00.000Z",
    },
    Object {
      "EventId": 1,
      "User": Object {
        "id": 1,
        "imageAvatar": null,
        "username": "pioluk",
      },
      "UserId": 1,
      "createdAt": "2016-12-03T12:30:00.000Z",
      "deletedAt": null,
      "id": 2,
      "text": "Dolor Amet",
      "updatedAt": "2016-12-03T12:30:00.000Z",
    },
  ],
  "error": null,
  "isLoading": false,
  "totalCount": 10,
}
`;

exports[`comments reducer should handle REMOVE_COMMENT 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": null,
  "isLoading": false,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle REMOVE_COMMENT_FAILURE 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": [Error: Error removing comment],
  "isLoading": false,
  "totalCount": 0,
}
`;

exports[`comments reducer should handle REMOVE_COMMENT_SUCCESS 1`] = `
Object {
  "comments": Array [
    Object {
      "EventId": 1,
      "UserId": 1,
      "createdAt": "2016-12-03T12:30:00.000Z",
      "deletedAt": null,
      "id": 2,
      "text": "Dolor Amet",
      "updatedAt": "2016-12-03T12:30:00.000Z",
      "user": Object {
        "createdAt": "2016-12-01T13:02:38.213Z",
        "email": "181423@edu.p.lodz.pl",
        "id": 1,
        "imageAvatar": null,
        "updatedAt": "2016-12-01T13:02:38.213Z",
        "username": "pioluk",
      },
    },
  ],
  "error": null,
  "isLoading": false,
}
`;

exports[`comments reducer should return the initial state 1`] = `
Object {
  "commentCount": 0,
  "comments": Array [],
  "error": null,
  "isLoading": false,
  "totalCount": 0,
}
`;
